add a default bounce expiration

ssalinas opened the pull request:
Don't allow a bounce that will go on forever. Defaults to 60 minutes and shows in the ui as:
![screen shot 2016-01-28 at 9 14 55 am](https://cloud.githubusercontent.com/assets/6034439/12646702/9fe22d36-c59f-11e5-99b0-cea1cfa947f9.png)

wsorenson commented on SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java:
```diff
@@ -171,34 +190,18 @@ public SingularityRequestParent bounce(@ApiParam("The request ID to bounce") @Pa
-      actionId = bounceRequest.get().getActionId();
-      message = bounceRequest.get().getMessage();
-
-      if (bounceRequest.get().getDurationMillis().isPresent() && !actionId.isPresent()) {
```
This logic was changed

ssalinas replied:
Thanks, I'll check against staging to see what changed

wsorenson replied:
Ah, now that I look again - are we now saying it's impossible to have unlimited bounce?

If so, what will you do if someone enters a negative value for this configuration? Need to have validation on it or allow people to set it to 0/negative to indicate that they don't want to use this feature.

In which case, we'd only set actionId if there was a duration set.

ssalinas replied:
Do you think we should still create an expiring object either way? One thing I realized in writing this is that if we don't create the expiring object, there isn't actually a way to cancel the bounce (without doing a janky pause/unpause)

wsorenson replied:
Good point, maybe just setup the config to validate that this must be non-zero, positive, then it's not an issue. I think you can do that with annotations. Someone could set it really high.

wsorenson commented on SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java:
```diff
@@ -162,7 +163,25 @@ public SingularityRequestParent bounce(@ApiParam("The request ID to bounce") @Pa
+      bounceRequest = new SingularityBounceRequest(
+        Optional.<Boolean>absent(),
+        Optional.<Boolean>absent(),
+        Optional.of(TimeUnit.MINUTES.toMillis(configuration.getDefaultBounceExpirationMinutes())),
```
Instead of storing this in the object, why not check it later? Recreating the object like this is gross.

wsorenson replied:
Also, as a general rule, late-binding of configuration properties makes testing easier.

ssalinas commented:
Ok, reworked this a bit. Not reconstructing the object anymore, just getting the `durationMillis` slightly differently for bounces in `SingularityExpiringUserActionPoller` so we can use the default and added a `@Min` for the config value. Tweaked the UI to show default value as well if there isn't a duration present for bounce
